### PR TITLE
Implement surface annotation and improve roi preferences

### PR DESCRIPTION
**Pull request by wadjih-bencheikh18**
closes : https://github.com/zakodium-oss/pixelium/issues/50

### CODE REVIEW

File: src/components/roi/annotation/ConvexHullAnnotation.tsx [hunk 1]
```diff
@@ -17,9 +17,9 @@ function ConvexHullAnnotation({ roi }: ConvexHullAnnotationProps) {
   );
 
   const preferences = usePreferences();
-  const color = useMemo(
-    () => preferences.rois.annotations.convexHull.color,
-    [preferences.rois.annotations.convexHull.color],
+  const { color, enabled, fontColor, fontSize } = useMemo(
+    () => preferences.rois.annotations.convexHull,
+    [preferences.rois.annotations.convexHull],
```
Comment (stropitek) [inline src/components/roi/annotation/ConvexHullAnnotation.tsx:22]: The `useMemo` can be removed. It has no effect.

File: src/components/roi/annotation/FeretAnnotation.tsx [hunk 2]
```diff
@@ -18,46 +18,68 @@ function FeretAnnotation({ roi }: FeretAnnotationProps) {
...
     [maxDiameter, minDiameter],
   );
 
   const preferences = usePreferences();
-  const color = useMemo(
-    () => preferences.rois.annotations.feretDiameters.color,
-    [preferences.rois.annotations.feretDiameters.color],
+  const { color, enabled, fontColor, fontSize } = useMemo(
+    () => preferences.rois.annotations.feretDiameters,
+    [preferences.rois.annotations.feretDiameters],
```
Comment (stropitek) [inline src/components/roi/annotation/FeretAnnotation.tsx:36]: ditto

File: src/components/roi/annotation/MBRAnnotation.tsx [hunk 3]
```diff
@@ -16,9 +16,9 @@ function MBRAnnotation({ roi }: MBRAnnotationProps) {
     [roi.mbr.points],
   );
   const preferences = usePreferences();
-  const color = useMemo(
-    () => preferences.rois.annotations.minimalBoundingRectangle.color,
-    [preferences.rois.annotations.minimalBoundingRectangle.color],
+  const { color, enabled, fontSize, fontColor } = useMemo(
+    () => preferences.rois.annotations.minimalBoundingRectangle,
+    [preferences.rois.annotations.minimalBoundingRectangle],
```
Comment (stropitek) [inline src/components/roi/annotation/MBRAnnotation.tsx:21]: ditto

File: src/components/roi/annotation/SurfaceAnnotation.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  roi: Roi;
+}
+
+function SurfaceAnnotation({ roi }: SurfaceAnnotationProps) {
+  const preferences = usePreferences();
+
+  const { color, enabled } = useMemo(
+    () => preferences.rois.annotations.surface,
+    [preferences.rois.annotations.surface],
+  );
```
Comment (stropitek) [inline src/components/roi/annotation/SurfaceAnnotation.tsx:16]: ditto

File: src/components/roi/annotation/ConvexHullAnnotation.tsx (hunk 1 above)
Comment (wadjih-bencheikh18) [reply src/components/roi/annotation/ConvexHullAnnotation.tsx]: fixed, thank you

**Comment by lpatiny** [comment]
We should make the distinction between just 'show' and 'display value'. We may hide the textual value while still keeping the annotation. Those are 2 different checkboxes.

I would add one more column in front called 'Display' that would show / hide the annotation (indenpendent from the text value)

image

**Comment by wadjih-bencheikh18** [comment]
> he way it is implemented being drawing thousands of little rectangle.

Could you please suggest another way to do it, I couldn't extract borders only to draw polygon.

**Comment by wadjih-bencheikh18** [comment]
> I would add one more column in front called 'Display'

if display=false and displayValue=true do we show the textual values?

**Comment by lpatiny** [comment]
Yes

**Review by lpatiny — APPROVED** [review]
Should be improved in the future because surface annotation is too slow but this is another issue: https://github.com/zakodium-oss/pixelium/issues/60

**Comment by lpatiny** [comment]
@stropitek This PR can be merge for me if the code is ok.
Concerning the check that fails should this be removed with the message of the merge commit ?

**Pull request merged**